src/app/product/product.service.ts

```diff
@@ -0,0 +1,12 @@
+import { Injectable } from "@angular/core";
+
+@Injectable({
+    providedIn: "root",
+  })
+  export class ProductService {
+    getInfoOfQuantite(quantite: number, product: string) {
+      return quantite > 10
+        ? quantite + " " + product
+        : "commander des " + product;
+    }
+  }
```

Comment:
order, cancel, claim methods should belong here
also extract the base api url to the environement.ts files, to make it easier to switch between api based on env (test, dev, staging, prod...Etc)
```ts
  export class ProductService {

    constructor(private http: HttpClient) { }

    getAll(): Observable<Product[]>{
      const url = `${environment.apiBaseUrl}/products`;
      return this.htto.get<Product[]>(url);
    }

    order(model: OrderModel): Observable<OrderResponse>{
      const url = `${environment.apiBaseUrl}/products/${model.id}/order`;
      return this.http.post<Product[]>(url, model);
    }

   // ... etc
  }

```